File: java/src/main/java/com/android/widget/RecyclerView/AATest/TestSingleAdapterActivity.java
package com.android.widget.RecyclerView.AATest;

import android.os.Bundle;

import com.android.frame.mvc.BaseActivity;
import com.android.java.databinding.ActivityTestAdapterBinding;

import java.util.ArrayList;
import java.util.List;

/**
 * Create by xuzhb on 2020/1/21
 * Desc:单一布局
 */
public class TestSingleAdapterActivity extends BaseActivity<ActivityTestAdapterBinding> {

    private String mSelectedItem;
    private List<String> mList;
    private TestSingleAdapter mAdapter;

    @Override
    public void handleView(Bundle savedInstanceState) {
        mList = createData();
        mAdapter = new TestSingleAdapter(this, mList, mSelectedItem);
        binding.srl.setEnabled(false);
        binding.rv.setAdapter(mAdapter);
    }

    private List<String> createData() {
        List<String> list = new ArrayList<>();
        list.add("工商银行");
        list.add("招商银行");
        list.add("农业银行");
        list.add("建设银行");
        list.add("中国银行");
        list.add("广发银行");
        list.add("交通银行");
        list.add("平安银行");
        list.add("兴业银行");
        list.add("民生银行");
        return list;
    }

    @Override
    public void initListener() {
        mAdapter.setOnItemClickListener((data, position) -> {
            mSelectedItem = (String) data;
            mAdapter.setData(mList, mSelectedItem);
        });
    }

}

File: kotlin/src/main/java/com/android/widget/RecyclerView/AATest/TestSingleAdapterActivity.kt
package com.android.widget.RecyclerView.AATest

import android.os.Bundle
import com.android.basicproject.databinding.ActivityTestAdapterBinding
import com.android.frame.mvc.BaseActivity

/**
 * Created by xuzhb on 2019/10/31
 * Desc:单一布局
 */
class TestSingleAdapterActivity : BaseActivity<ActivityTestAdapterBinding>() {

    private var mSelectedItem = ""
    private var mList: MutableList<String> = mutableListOf()
    private val mAdapter by lazy { TestSingleAdapter(this, mList, mSelectedItem) }

    override fun handleView(savedInstanceState: Bundle?) {
        mList = createData()
        binding.srl.isEnabled = false
        binding.rv.adapter = mAdapter
    }

    private fun createData(): MutableList<String> = mutableListOf(
        "工商银行", "招商银行", "农业银行", "建设银行", "中国银行", "广发银行", "交通银行",
        "平安银行", "兴业银行", "民生银行"
    )

    override fun initListener() {
        mAdapter.setOnItemClickListener { obj, position ->
            mSelectedItem = obj as String
            mAdapter.setData(mList, mSelectedItem)
        }
    }

}